### DISCUSSION TITLE
RSC and Server action with Websocket

### DISCUSSION THREAD

**Discussion by brandfactoryitalia** [vercel/next.js#59242] [Ideas]
### Goals

1) Allow updating of component servers on the client from an event triggered exclusively on the backend (example: when the database is updated)

2) Allow the client to receive a server action sent by an event triggered exclusively on the backend

### Background

Currently the only ways I know are to establish a websocket connection with the client, and either pass it the data to update and do it manually... or every time the client receives an update from the websocket it calls a server action which, via revalidatePath, updates everything

### Proposal

.

**Comment by icyJoseph** [comment]
I remember on the JS Party podcast, the React team sort of talked about this:

- https://changelog.com/jsparty/267#t=2905

> But I know one of the things we’ve talked about – so for example, you can wait for a single piece of data to resolve. You can do an async/await, and – one of the things that we’re exploring is support for like an async iterable, so that you can actually await for a list of values to resolve, and then stream those down. So that kind of addresses it, but that is the sort of list of items use case, where you want to just like load one after the other, and kind of incrementally stream them in. If you’re talking about things like WebSockets, where you want to just have kind of like a Pub/Sub system that’s like sending you information, that right now would be kind of done as a client component.

I also answered a bit here: https://github.com/vercel/next.js/discussions/56750#discussioncomment-7264214
